backend/src/main/java/com/example/backend/service/TripService.java

```diff
@@ -29,7 +29,9 @@ public Trip getById(String id) throws Exception {
 
     public Trip add(Trip trip) {
         for (Location location : trip.getLocations()) {
-            location.setId(this.idGenerator.generateRandomId());
+            if (location.getId() == null) {
```

Comment:
Is that check necessary? Would we access locations that have IDs already set?

Reply:
Ja, beim Update haben bestehende Locations eine ID - die darf hier nicht überschrieben werden.